### PR TITLE
[core][Qt] Use Qt for UTF16, since `codecvt` is not always available

### PR DESCRIPTION
**Pull request by tmpsantos**
Instead, use Qt's codec.

### CODE REVIEW

File: platform/qt/config.cmake [hunk 1]
```diff
@@ -50,6 +50,12 @@ macro(mbgl_platform_core)
         target_sources(mbgl-core PRIVATE platform/qt/src/bidi.cpp)
     endif()
 
+    if(WITH_QT_4)
+        target_sources(mbgl-core PRIVATE platform/default/utf.cpp)
+    else()
+        target_sources(mbgl-core PRIVATE platform/qt/src/utf.cpp)
```
Comment (tmpsantos) [inline platform/qt/config.cmake]: `QString::fromStdString(...).toStdU16String();` is only available for Qt5+

File: platform/qt/src/string_stdlib.cpp [hunk 2]
```diff
@@ -9,13 +9,13 @@ namespace mbgl {
 namespace platform {
 
 std::string uppercase(const std::string& str) {
```
Comment (tmpsantos) [inline platform/qt/src/string_stdlib.cpp]: @brunoabinader I added this change after you reviewed. Do you mind having a look again?

Comment (tmpsantos) [inline platform/qt/src/string_stdlib.cpp]: The context is we should never use c_str() for multibyte.

Comment (brunoabinader) [inline platform/qt/src/string_stdlib.cpp]: > The context is we should never use c_str() for multibyte.

Interesting - I wasn't aware of that! 👍

Comment (ChrisLoer) [inline platform/qt/src/string_stdlib.cpp]: > The context is we should never use c_str() for multibyte.

That is interesting -- I thought the only difference was that `data()` could leave off null termination (and probably for most of our compiler targets it null terminates anyway, which makes it easy to introduce sneaky bugs depending on the null termination).

**Pull request merged**